### ISSUE TITLE
Prisma migrate error in v2.1.0 when using dbgenerated() default on a Json field

### ISSUE THREAD

**Issue by jwld** [prisma/migrate#488]
## Bug description

Report id = 8028

Using `dbgenerated()` on a `Json` field throws an error when running `prisma migrate up`.

This is only broken in 2.1.0, it works fine in 2.0.1.

The generated SQL in the README shows this:
```
CREATE TABLE "default$default"."Service" (
"formula" jsonb  NOT NULL DEFAULT ,"id" text  NOT NULL ,
    PRIMARY KEY ("id"))
```

Whereas in v2.0.1 it produces:
```
CREATE TABLE "default$default"."Service" (
"formula" jsonb  NOT NULL ,"id" text  NOT NULL ,
    PRIMARY KEY ("id"))
```

## How to reproduce

1. Create schema:
```
model Service {
  formula Json   @default(dbgenerated())
  id      String @default(cuid()) @id
}
```

2. Run `prisma migrate save` and `prisma migrate up` - throws the following:
```
Failure during a migration command: Connector error. (error: Error querying the database: Error querying the database: db error: ERROR: syntax error at or nea
r ","
   0: migration_core::api::ApplyMigration
           with migration_id="20200624154249-f"
             at migration-engine/core/src/api.rs:77)
```

## Expected behavior

Able to use `dbgenerated` on Json fields as before.

## Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Prisma version: 2.1.0
- Node.js version: 14.2.0

**Comment by tomhoule** [comment]
Thanks for reporting this! This bug was fixed earlier today on the main branch: https://github.com/prisma/prisma-engines/pull/845

The fix should already be in the last `dev` release, and it will be in 2.2.0.

Please also note that `@default(dbgenerated())` is purely a workaround for introspection, these defaults are ignored by `prisma migrate`.

**Comment by jwld** [comment]
@tomhoule  Thanks!

Am I right in thinking we still need the `@default(dbgenerated())` so that the generated prisma client will respect the default set on the DB, or can it be removed / is there a better way currently?

**Comment by tomhoule** [comment]
Ah I just noticed it was on a Json field. There was a separate bug that made it the only viable workaround there (`@default` made migrate crash on `Json` fields), but this is also going to be fixed very very soon, there's an open PR here: https://github.com/prisma/prisma-engines/pull/851

Once this is merged, you can either leave `@default(dbgenerated())` or write your JSON default as a string (something like `@default("{\"abc\": true}")`.

**Comment by jwld** [comment]
That all sounds great - thanks!

**Comment by jwld** [comment]
Fixed in 2.2
